state/indexer-service.md

```diff
@@ -0,0 +1,58 @@
+# Indexer Service
+
+## Abstract
+
+The Indexer service indexes transactions and blocks using events emitted by the block executor that can later be used to build services like block explorers that ingest this data.
+
+## Protocol/Component Description
+
+The Indexer service consists of three main components: an event bus, a transaction indexer and a block indexer.
+
+### Event Bus
+
+The event bus is a messaging service between the full node and the indexer service. It serves events emitted by block executor of the full node to the indexer service where events are routed to the relevant indexer component based on type.
+
+### Block Indexer
+
+The [Block Indexer][block_indexer] indexes BeginBlock and EndBlock events with an underlying KV store. Block events are indexed by their height, such that matching search criteria returns the respective block height(s).
```

Comment:
I think this changed with v.0.50 @gupadhyaya  right? We don't have begin and endblock anymore but finalize block instead.

Reply:
The block indexer still indexes the `BeginBlock` and `EndBlock` events: https://github.com/rollkit/rollkit/blob/05ff97ea78862631436c8b638d50a63de0e1b7b0/state/indexer/block/kv/kv.go#L57